packages-nextgen/kestrel_datasource_opensearch/src/kestrel_datasource_opensearch/ossql.py

```diff
@@ -116,14 +117,18 @@ def _render_exp(self, exp: BoolExp) -> str:
             rhs = _render_comp(exp.rhs)
         return _and(lhs, rhs) if exp.op == ExpOp.AND else _or(lhs, rhs)
 
+    def add_DataSource(self, ds: DataSource) -> None:
```

Comment:
Suggest to remove this method and use the constructor to handle all table/index related info.